src/wallet/wallet2.cpp

```diff
@@ -4245,6 +4443,447 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+    const uint64_t total_received = t.total_received;
+    uint64_t total_sent = t.total_sent;
+
+    // Check key images - subtract fake outputs from total_sent
+    for(const auto &so: t.spent_outputs)
+    {
+      crypto::public_key tx_public_key;
+      crypto::key_image key_image;
+      THROW_WALLET_EXCEPTION_IF(string_tools::validate_hex(64, so.tx_pub_key), error::wallet_internal_error, "Invalid tx_pub_key field");
+      THROW_WALLET_EXCEPTION_IF(string_tools::validate_hex(64, so.key_image), error::wallet_internal_error, "Invalid key_image field");
+      string_tools::hex_to_pod(so.tx_pub_key, tx_public_key);
+      string_tools::hex_to_pod(so.key_image, key_image);
+
+      if(!light_wallet_key_image_is_ours(key_image, tx_public_key, so.out_index)) {
+        THROW_WALLET_EXCEPTION_IF(so.amount > t.total_sent, error::wallet_internal_error, "Lightwallet: total sent is negative!");
+        total_sent -= so.amount;
+      }
+    }
+
+    // Do not add tx if empty. 
+    if(total_sent == 0 && total_received == 0)
+      continue;
+    
+    crypto::hash payment_id = null_hash;
+    crypto::hash tx_hash;
+    
+    THROW_WALLET_EXCEPTION_IF(string_tools::validate_hex(64, t.payment_id), error::wallet_internal_error, "Invalid payment_id field");
+    THROW_WALLET_EXCEPTION_IF(string_tools::validate_hex(64, t.hash), error::wallet_internal_error, "Invalid hash field");
+    string_tools::hex_to_pod(t.payment_id, payment_id);
+    string_tools::hex_to_pod(t.hash, tx_hash);
+
+    // lightwallet specific info
+    bool incoming = (total_received > total_sent);
+    address_tx address_tx;
+    address_tx.m_tx_hash = tx_hash;
+    address_tx.m_incoming = incoming;
+    address_tx.m_amount  =  incoming ? total_received - total_sent : total_sent - total_received;
+    address_tx.m_block_height = t.height;
+    address_tx.m_unlock_time  = t.unlock_time;
+    address_tx.m_timestamp = t.timestamp;
+    address_tx.m_coinbase  = t.coinbase;
+    address_tx.m_mempool  = t.mempool;
+    m_light_wallet_address_txs.emplace(tx_hash,address_tx);
+
+    // populate data needed for history (m_payments, m_unconfirmed_payments, m_confirmed_txs)
+    // INCOMING transfers
+    if(total_received > total_sent) {
+      payment_details payment;
+      payment.m_tx_hash = tx_hash;
+      payment.m_amount       = total_received - total_sent;
+      payment.m_block_height = t.height;
+      payment.m_unlock_time  = t.unlock_time;
+      payment.m_timestamp = t.timestamp;
+        
+      if (t.mempool) {   
+        if (std::find(unconfirmed_payments_txs.begin(), unconfirmed_payments_txs.end(), tx_hash) == unconfirmed_payments_txs.end()) {
+          pool_txs.push_back(tx_hash);
+          m_unconfirmed_payments.emplace(tx_hash, payment);
+          if (0 != m_callback) {
+            cryptonote::transaction dummy_tx;
+            m_callback->on_unconfirmed_money_received(t.height, payment.m_tx_hash, dummy_tx, payment.m_amount);
```

Comment:
If nothing uses them, and if you don't have access to them, then I guess removing them is fine, and if something needs to use them later, we can do what you suggest and have separate calls.